bids/variables/io.py

```diff
@@ -119,8 +120,18 @@ def _load_time_variables(layout, dataset=None, columns=None, scan_length=None,
     if dataset is None:
         dataset = NodeIndex()
 
-    images = layout.get(return_type='file', type='bold', modality='func',
-                        extensions='.nii.gz', **selectors)
+    if _settings['loop_preproc']:
+        selectors['type'] = 'preproc'
+        # Select any space, to only loop over each run once
```

Comment:
Is it safe to assume that if there are multiple spaces, there are multiple redundant images for each one? I.e., couldn't there be weird cases where some images are only in one space, and some are only in another? This is probably enough of an edge case that we don't need to deal with it right now, but maybe maybe a note in a comment.